src/hybrid/parse/hybrid_callbacks.c

```diff
@@ -0,0 +1,404 @@
+            QueryError_SetError(status, QUERY_EPARSEARGS, "AS needs argument");
+            goto error;
+        }
+        stp->base.alias = rm_strndup(alias, aliasLen);
+    } else {
+        stp->base.alias = rm_strndup(expr, exprLen);
+    }
+    return;
+
+error:
+    if (stp) {
+        AGPLN_PopStep(&stp->base);
+        stp->base.dtor(&stp->base);
+    }
+}
+
+// LOAD callback - implements EXACT original logic from handleLoad
+void handleLoad(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    ctx->specifiedArgs |= SPECIFIED_ARG_LOAD;
+    ArgsCursor *ac = parser->cursor;  // Get remaining args from parser cursor
+    QueryError *status = ctx->status;
+
+    // Get the first argument from the string value
+    const char *firstArg = *(const char**)value;
+    ArgsCursor loadfields = {0};
+    if (strcmp(firstArg, "*") == 0) {
+        // Successfully got a '*', load all fields
+        REQFLAGS_AddFlags(ctx->reqFlags, QEXEC_AGG_LOAD_ALL);
+    } else {
+        // Try to parse the first argument as a number of fields to load
+        char *end = NULL;
+        long long numFields = strtoll(firstArg, &end, 10);
+        if (*end != '\0' || numFields <= 0) {
+            QueryError_SetError(status, QUERY_EPARSEARGS, "Bad arguments for LOAD: Expected number of fields or `*`");
+            return;
+        }
+        // Successfully got a number, slice that many fields
+        if (AC_GetSlice(ac, &loadfields, numFields) != AC_OK) {
+            QueryError_SetError(status, QUERY_EPARSEARGS, "Not enough arguments for LOAD");
+            return;
+        }
+    }
+
+    PLN_LoadStep *lstp = rm_calloc(1, sizeof(*lstp));
+    lstp->base.type = PLN_T_LOAD;
+    lstp->base.dtor = loadDtor;
+    if (loadfields.argc > 0) {
+        lstp->args = loadfields;
+        lstp->keys = rm_calloc(loadfields.argc, sizeof(*lstp->keys));
+    }
+
+    if (*ctx->reqFlags & QEXEC_AGG_LOAD_ALL) {
+        lstp->base.flags |= PLN_F_LOAD_ALL;
+    }
+
+    AGPLN_AddStep(ctx->plan, &lstp->base);
+}
+
+// FILTER callback - implements EXACT original logic from handleApplyOrFilter with isApply=0
+void handleFilter(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    QueryError *status = ctx->status;
+    ctx->specifiedArgs |= SPECIFIED_ARG_FILTER;
+
+    // Get the expression from the string value
+    const char *expr = *(const char**)value;
+    size_t exprLen = strlen(expr);
+
+    HiddenString* expression = NewHiddenString(expr, exprLen, false);
+    PLN_MapFilterStep *stp = PLNMapFilterStep_New(expression, PLN_T_FILTER);
+    HiddenString_Free(expression, false);
+    AGPLN_AddStep(ctx->plan, &stp->base);
+}
+
+// TIMEOUT callback - implements EXACT original logic from handleTimeout
+void handleTimeout(ArgParser *parser, const void *value, void *user_data) {
```

Comment:
why are there no actual implementation here? not yet implemented? is it a known limitation or debt?